subsys/random/rand32_timer.c

```diff
@@ -62,20 +62,16 @@ uint32_t z_impl_sys_rand32_get(void)
 
 void z_impl_sys_rand_get(void *dst, size_t outlen)
 {
-	uint32_t len = 0;
-	uint32_t blocksize = 4;
+	uint8_t *udst = (uint8_t *)dst;
+	uint32_t blocksize;
 	uint32_t ret;
-	uint32_t *udst = (uint32_t *)dst;
 
-	while (len < outlen) {
+	while (outlen) {
 		ret = sys_rand32_get();
-		if ((outlen-len) < sizeof(ret)) {
-			blocksize = len;
-			(void)memcpy(udst, &ret, blocksize);
-		} else {
-			(*udst++) = ret;
-		}
-		len += blocksize;
+		blocksize = outlen >> 2 ? 4 : outlen & 0x3;
```

Comment:
`outlen & 0x3` is redundant when `outlen` is less than 4.
Add paranthesis and use `sizeof(ret)`

```suggestion
		blocksize = (outlen >> 2) ? sizeof(ret) : outlen;
```

Reply:
Using MIN would make it clearer IMO.
```suggestion
		blocksize = MIN(outlen , sizeof(ret)); 
```